internal/custom-structs.go

```diff
@@ -147,53 +149,79 @@ func buildNotification(db *sql.DB, vt string, p Play, e PlayEvent, teams GameTea
 
 	tweet.MainTweet = fmt.Sprintf(`%v\n%v (%v)\n%v %v - %v @ %v\nCount Now %v - %v, %v %v`, thead, player.FullName, t, p.About.HalfInning, p.About.Inning, awayTeam, homeTeam, e.Count.Balls, e.Count.Strikes, e.Count.Outs, o)
 
+	// move to a function
 	pluralization := "'"
 	if player.FullName[len(player.FullName)-1:] != "s" {
 		pluralization = fmt.Sprintf(`%ss`, pluralization)
 	}
 
-	if p.About.HalfInning == "top" {
-		teamAbbr = teams.Home.ClubName
-	} else {
-		teamAbbr = homeTeam
+	suffix := getOrdinalSuffix(pt)
+
+	var dailyNote string
+	if ptt > 1 {
+		dailyNote = fmt.Sprintf(" (%d%s today)", ptt, getOrdinalSuffix(ptt))
+		fmt.Printf("DAILY NOTE!!\n%s\n", dailyNote)
 	}
 
-	tweet.ReplyTweet = fmt.Sprintf(`That's %s%s %d%s violation this season. As a team the %s have %d violations this year (%d %s violations)`, player.FullName, pluralization, pt, "th", teamAbbr, tt, ttt, strings.ToLower(a))
+	tweet.ReplyTweet = fmt.Sprintf(`That's %s%s %d%s violation this season%s. As a team the %s have %d violations this year (%d %s violations)`, player.FullName, pluralization, pt, suffix, dailyNote, teamAbbr, tt, ttt, strings.ToLower(a))
 
 	return tweet
 }
 
-func saveViolation(db *sql.DB, code string, p Player, team string) (int, int, int) {
-	stmt, err := db.Prepare("INSERT INTO player_stats(player_id, player_name, team_name, code, date) VALUES(?, ?, ?, ?, NOW())")
+func getOrdinalSuffix(i int) string {
+	j := i % 10
+	k := i % 100
+	if j == 1 && k != 11 {
+		return "st"
+	}
+	if j == 2 && k != 12 {
+		return "nd"
+	}
+	if j == 3 && k != 13 {
+		return "rd"
+	}
+	return "th"
+}
+
+func saveViolation(db *sql.DB, code string, p Player, team string, date string) (int, int, int, int) {
+	table := os.Getenv("DB_TABLE")
+	stmt, err := db.Prepare(fmt.Sprintf("INSERT INTO %s(player_id, player_name, team_name, code, date) VALUES(?, ?, ?, ?, ?)", table))
```

Comment:
It doesn't really matter to the impact or quality of the code, but because you are using placeholders here, Go would automatically prepare the statement. It also doesn't change the performance at all. You could validate this by using Wireshark or something and watching the traffic to the database, it wouldn't change between the prepare + exec, versus an Exec with the statement with placeholders. The same network traffic will end up being sent.

See: https://dev.mysql.com/doc/refman/8.0/en/sql-prepared-statements.html

In particular...

> A prepared statement is specific to the session in which it was created. If you terminate a session without deallocating a previously prepared statement, the server deallocates it automatically.
>
> A prepared statement is also global to the session. If you create a prepared statement within a stored routine, it is not deallocated when the stored routine ends.

So that's what the server will do.

Now, if you were to retrieve a connection, and run the same queries, you could use the Prepare API, and reuse the `driver.Stmt` that is returned to make sure you doesn't get automatic deallocations depending on if the driver you're using deallocates the automatic prepared statements immediately after use because perhaps they don't name the statements, and they are an unnamed prepared statement, which only stays prepared until the next statement. The Postgres driver lib/pq iirc does this.